app/client/jsx/ChatStreamRoom.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect } from "react";
+
+/**
+ * This component is used to render a Converse.js chat component on the
+ * left side and an IFrame on the right. While this component does have
+ * the jitsi-video class in the render it is only for placement: there
+ * is no jitsi video in this component.
+ *
+ * This component works in conjunction with the CHATSTREAM room type.
+ *
+ * @param props a combo of roomData and jitsiData
+ * @returns
+ */
+export const ChatStreamRoom = ({
+  id: roomId,
+  name: roomName,
+  displayName,
+  boshUrl,
+  iframeOptions: { src },
+}) => {
+  // register the 'jitsi-plugin' and initialize converse and cleanup after close
+  useEffect(() => {
+    let logout = null;
+    let plugins = null;
+
+    // configure the 'jitsi-plugin' to get our hands on the converse api logout
+    try {
+      window.converse.plugins.add("jitsi-plugin", {
+        initialize: function () {
+          logout = this._converse.api.user.logout;
+          plugins = this._converse.pluggable.plugins;
+
+          // add event listener to restucture the chatbox
+          this._converse.api.listen.on("chatBoxInsertedIntoDOM", () => {
```

Comment:
Apparently the `chatBoxInsertedIntoDOM` event will be removed in the next major version: https://github.com/conversejs/converse.js/blob/master/CHANGES.md

We could replace it with one of these events: https://github.com/conversejs/converse.js/blob/master/src/plugins/minimize/index.js#L128-L133